railties/lib/rails/generators.rb

```diff
@@ -29,6 +29,7 @@ module Generators
     autoload :TestCase,        "rails/generators/test_case"
 
     mattr_accessor :namespace
+    mattr_reader :generated_files, default: []
```

Comment:
Hm, not liking exposing this here. Can we manage everything entirely within `add_generated_file`? Maybe via `(@@generated_files ||= []) << file` or something?

If not, I'd move the `_reader` above the `_accessor` since that's the style we follow, iirc.

Reply:
I fixed that manage everything entirely within `add_generated_file`.